app/models/validations/sales/financial_validations.rb

```diff
@@ -52,6 +52,20 @@ def validate_percentage_bought_not_greater_than_percentage_owned(record)
     end
   end
 
+  def validate_percentage_bought_at_least_threshold(record)
+    return unless record.stairbought && record.type
+
+    if [2, 16, 18, 24].include? record.type
+      threshold = 10
+    elsif record.type == 30
+      threshold = 1
+    end
+
+    if threshold && record.stairbought < threshold
+      record.errors.add :stairbought, I18n.t("validations.financial.staircasing.percentage_bought_must_be_at_least_threshold", threshold:)
```

Comment:
I think we should also add an error message to `:type` (so that if you set a valid value for stairbought and then go back and change the shared ownership type such that the stairbought value is now invalid, then an error message is shown on the shared ownership type page.

Alternatively do we want to just clear the stairbought value if the shared ownership type changes? Presumably not.

Any other options?

Reply:
@SamSeed-Softwire Yes, I would add the message to type as well.